client/internal/config.go

```diff
@@ -196,20 +208,64 @@ func ReadConfig(input ConfigInput) (*Config, error) {
 			return nil, err
 		}
 		config.SSHKey = string(pem)
+		log.Infof("generated config.SSHKey")
 		refresh = true
 	}
 
-	if config.WgPort == 0 {
+	if config.WgPort == 0 && input.WgPort == 0 {
+		log.Infof("using default Wireguard Port: %#v ", iface.DefaultWgPort)
 		config.WgPort = iface.DefaultWgPort
 		refresh = true
 	}
```

Comment:
```suggestion
         if input.WgPort != 0 && input.WgPort != config.WgPort {
		log.Infof("new Wireguard Port provided, updated to %d (old value %d)", input.WgPort, config.WgPort)
		config.WgPort = input.WgPort
		refresh = true
	}
	// previous version config upgrade check because WgPort didn't exist
	if config.WgPort == 0 {
		log.Infof("using default Wireguard Port: %d", iface.DefaultWgPort)
		config.WgPort = iface.DefaultWgPort
		refresh = true
	}
```